### PR TITLE
Set up testing environment for API

### PR DESCRIPTION
**Pull request by viktorsommar**
### User Story
```
As a development team
In order to be able to make sure our code is doing what it is supposed to do
We would like to be able to write and run automated
```
## Changes proposed in this pull request:
* Adding setup for rspec, shoulda matchers and factory bot.
## What I have learned working on this feature:
* Setting up testing environment in rails
## Packages/Gems added
- rails-pry, rspec-rails, shoulda-matchers, factory-bot_rails

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -7,9 +7,14 @@ gem 'pg', '>= 0.18', '< 2.0'
 gem 'puma', '~> 4.1'
 gem 'bootsnap', '>= 1.4.2', require: false
 gem 'rack-cors', require: 'rack/cors'
+gem 'prettier'
```
Comment (emtalen) [inline Gemfile:10]: why have you added prettier?

File: app/channels/application_cable/channel.rb [hunk 2]
```diff
@@ -1,4 +1,3 @@
 module ApplicationCable
-  class Channel < ActionCable::Channel::Base
-  end
+  class Channel < ActionCable::Channel::Base; end
 end
```
Comment (emtalen) [inline app/channels/application_cable/channel.rb]: This looks weird, we don't use ; in ruby in that way

File: app/channels/application_cable/connection.rb [hunk 3]
```diff
@@ -1,4 +1,3 @@
 module ApplicationCable
-  class Connection < ActionCable::Connection::Base
-  end
+  class Connection < ActionCable::Connection::Base; end
 end
```
Comment (emtalen) [inline app/channels/application_cable/connection.rb:4]: Same goes here

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -1,2 +1 @@
-class ApplicationController < ActionController::API
-end
+class ApplicationController < ActionController::API; end
```
Comment (emtalen) [inline app/controllers/application_controller.rb]: And here

File: config/routes.rb [hunk 5]
```diff
@@ -1,2 +1 @@
-Rails.application.routes.draw do
-end
+Rails.application.routes.draw {}
```
Comment (emtalen) [inline config/routes.rb]: Why did you change this?

File: spec/rails_helper.rb [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+end
+
+RSpec.configure do |config|
+  config.fixture_path = "#{::Rails.root}/spec/fixtures"
+  config.use_transactional_fixtures = true
+  config.infer_spec_type_from_file_location!
+  config.filter_rails_from_backtrace!
+  config.include FactoryBot::Syntax::Methods
+  config.include(Shoulda::Matchers::ActiveRecord, type: :model)
+end
```
Comment (emtalen) [inline spec/rails_helper.rb:24]: add the json response helper as oliver said earlier

**Review by Odeane — COMMENTED** [review]
Seems right ..great work guys

File: Gemfile [hunk 7]
```diff
@@ -7,9 +7,14 @@ gem 'pg', '>= 0.18', '< 2.0'
...
```
Comment (oliverochman) [inline Gemfile]: why was this added?

**Review by emtalen — COMMENTED** [review]
Stay in scope of the chore/feature and only change the files that you need to.

**Review by oliverochman — CHANGES_REQUESTED** [review]
Please add the reaponse_json aupport file and configure it

**Review by Cma-B — APPROVED** [review]
good job!

File: Gemfile (hunk 7 above)
Comment (faraznaeem) [inline Gemfile]: Why?

File: Gemfile [hunk 8]
```diff
@@ -7,9 +7,14 @@ gem 'pg', '>= 0.18', '< 2.0'
...
 
 group :development, :test do
   gem 'byebug', platforms: [:mri, :mingw, :x64_mingw]
```
Comment (faraznaeem) [inline Gemfile:12]: You are using pry rails, no need for this
